debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/junit/MongoDbDatabaseProvider.java

```diff
@@ -36,7 +40,11 @@ private static MongoDbReplicaSet.Builder dockerReplicaSetBuilder() {
      * @return MongoDb Replica set
      */
     public static MongoDbReplicaSet dockerReplicaSet() {
-        return dockerReplicaSetBuilder().build();
+        var replicaSet = dockerReplicaSetBuilder();
+        if (NETWORK != null) {
+            replicaSet.network(NETWORK);
+        }
+        return replicaSet.build();
```

Comment:
Pass the network as an argument if needs be and create a non parameter overrride. We are not using a static filed like this.

Reply:
That's not working because of timing issues. The instance is created before we can pass the network if we don't sneak it in like this for now. 
This MongoDbDatabaseProvider and MongoDbDeployment etc needs bigger refactoring it seems. It is not possible to do it in an easy, clean way. Do you have a suggestion? Something which is achievable by tomorrow? That's the timeline.

Reply:
1) You don't actually need to use this class at all. It came to be due to series of refactorings and a need not to overhaul the entire existing test suite. 

2) That Version provider only really determines the version 
3) What seems to be the problem with `MongoDbDeployment`? 
4) I get the if, I just don't agree with it being based on the static attribute. I'd personally  change the 2 builder methods to just use the common network in case a null is passed to the builder.

Reply:
ad 1) What is the reason behind this class? I assume based on a system property you can switch the build to use external provided MongoDB/DocumentDB instance. If we want to support such tests, and I think it makes a lot of sense, it also makes sense to use this class.

ad 2) no, based on `mongodb.platform` system property you can switch to use an external instance, right?

ad 3) sorry, I meant `MongoDbPlatform`. It's used from `MongoDbDatabaseProvider` but `MongoDbPlatform` then calls code from `MongoDbDatabaseProvider` and there's no way to inject custom config.

ad 4) How you want to do that when using that abstraction layer that provides the switch for external instances? I think that's a possible temporary solution. 

I filed [DBZ-8017](https://issues.redhat.com/browse/DBZ-8017) for the proper refactoring.

Reply:
1) Centralise the "creation" of Mongo deployment FOR THE MOST COMMON USE CASES IN THE SUITE - which among others counts on usage WITH EMBEDDED ENGINE where you never have the problem with network in the first place. 

You absolutely don't have to use this class -- e.g. the auth tests are not using it, since obviously only those require the database to be set with authentication and it wouldn't make sense to rewrite everything.  

2) Yes. 

3) In case of external you don't even need the custom config. Your entire problem is the docker. 


Your entire problem boils down to the fact that you can't pass state down to the provider/supplier used in 


```
 public static MongoDbDeployment externalOrDockerReplicaSet() {
        var platform = MongoDbDatabaseVersionResolver.getPlatform();
        return platform.provider.get();
    }
``

So you can either create and override and refactor MongoDbPlatform. Or just don't use this method and have your test just work with `dockerReplicaSet()` (or add the ability to pass in the network). 

Look, I agree that this is something which in ideal world would be implemented differently -- however it is like this for historical reason where we just didn't have the resources for a complete test rewrite (something we will have to do in the future anyway). However we should make it worse and introducing a potentially unstable side effect does make it worse. 

E.g. where is the `NETWORK` attribute set back to null? How do you ensure that the tests which will run after / intertwined with yous will not suddenly stat using that network?

Reply:
see below. after you comment about the side-effects I incorporated the reset of NETWORK.